workers.py

```diff
@@ -117,3 +117,75 @@ def worker() -> None:
 
     thread = Thread(target=worker)
     thread.start()
+
+
+def topology_client_download(topology_client, entity: str) -> list:
+    """Download a list for a Topological Inventory Entity."""
+    return topology_client.query_path(entity)
+
+
+def download_topological_inventory_data(
+        topology_client,
+        source_id: str,
+        dest_url: str
+) -> None:
+    """Spawn a thread worker for data downloading Topological Inventory Data.
+
+    Requests the data to be downloaded and pass it to the next service
+    :param topology_client: Instance of TopologyInventoryClient
+           to assist downloading data
+    :param source_id: Data identifier
+    :param dest_url: Location where the collected data should be received
+    """
+    # When source_id is missing, create our own
+    source_id = source_id or str(uuid4())
+
+    def worker() -> None:
+        """Download, extract data and forward the content."""
+        thread = current_thread()
+        logger.debug('%s: Worker started', thread.name)
+
+        # Fetch data
+        entities = ["container_nodes",
+                    "volume_attachments",
+                    "volumes",
+                    "volume_types"]
+
+        topology_client_data = {}
+
+        for entity in entities:
+            prometheus_metrics.METRICS['gets'].inc()
+            try:
+                topology_client_data[entity] =\
+                    topology_client_download(topology_client, entity)
+                prometheus_metrics.METRICS['get_successes'].inc()
+            except requests.HTTPError as exception:
+                prometheus_metrics.METRICS['get_errors'].inc()
+                logger.error(
+                    '%s: Unable to fetch source data for "%s": %s',
+                    thread.name, source_id, exception
+                )
+                return
+
+        # Build the POST data object
+        data = {
+            'id': source_id,
+            'data': topology_client_data
```

Comment:
Why don't we create the `data` dict before and just append data into `data['data']`? What benefit has to create new variable with a longer name which does the same?